BarcodeDetector/BarcodeDetector_detect_method.html

```diff
@@ -0,0 +1,75 @@
+<div>{{draft}}{{securecontext_header}}{{DefaultAPISidebar("Barcode Detector API")}}</div>
+
+<p class="summary">The <strong><code>detect()</code></strong> method of the {{domxref("BarcodeDetector")}} interface returns a {{jsxref('Promise')}} which fulfills with an {{jsxref('Array')}} of detected barcodes within an image.</p>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox notranslate">var <var>detectedBarcode</var> = BarcodeDetector.detect(ImageBitmapSource);</pre>
+
+<h3 id="Parameters">Parameters</h3>
+
+<dl>
+  <dt><em>image</em></dt>
+  <dd>Receives an <code>ImageBitmapSource</code> as a parameter. This can be an {{HTMLElement('img', '<img> element')}}, a {{domxref('Blob')}} of type image or an {{domxref('ImageData')}} object.</dd>
+</dl>
+
+<h3 id="Returns">Return value</h3>
+
+<p>Returns a {{domxref('Promise')}} which fulfills with an array of <code>detectedBarcode</code> objects with the following properties:</p>
+
+<ul>
+  <li><code>boundingBox</code>: A {{domxref('DOMRectReadOnly')}}, which returns the dimensions of a rectangle representing the extent of a detected barcode, aligned with the image.</li>
+  <li><code>cornerPoints</code>: The x and y co-ordinates of the four corner points of the detected barcode relative to the image, starting with the top left and working clockwise. This may not be square due to perspective distortions within the image.</li>
+  <!-- EDIT & ADD A LINK ON PUBLISH -->
+  <li><code>format</code>: The detected barcode format. (For a full list of formats see the [landing page])</li>
+  <li><code>rawValue</code>: A {{jsxref('String')}} decoded from the barcode data.</li>
+</ul>
+
+<h3 id="Exceptions">Exceptions</h3>
+
+<!-- THIS IS NOT IN THE SPEC -->
+<dl>
+
+  <dt><code>TypeError</code></dt>
+
+  <dd>No parameter is specified or the <code>type</code> is not that of an <code>ImageBitmapSource</code>.</dd>
+
+</dl>
```

Comment:
If this is non-normative, then other browsers could throw different errors. Can you confirm that with Reilly? We may need to say that the error varies by browser.